### PR TITLE
debaml(p1s1): retain prompt/template data in bamlparser

### PR DESCRIPTION
**Pull request by invakid404**
## de-BAML Phase 1, Slice 1 — parser retention only

Extends baml-rest's own native parser (`bamlutils/bamlparser`) to **retain** static prompt/template representation. This is representation-only: config extraction is byte-for-byte unchanged, and there is **no serving/codegen change** (no descriptor builder, no `cmd/introspect` output change, no request-path consumer). Slice 2 (the build-only prompt descriptor sidecar) is intentionally out of scope.

### What changed

**`FunctionBlock`** gains a semantic projection of the prompt field:
- `PromptRaw *string` and `HasPrompt bool`, filled during post-parse normalization by scanning the already-normalized `Fields` in source order and taking the **final** `prompt` field (last-field-wins, matching BAML).
- A final **non-raw** prompt sets `HasPrompt` but leaves `PromptRaw` nil — an earlier raw prompt is never resurrected.
- The ordered `Fields` slice is untouched and remains the lossless record.

**`TemplateBlock`** replaces the name-only node with retained data:
- `Keyword` (`template_string` or the tolerated `string_template` alias, retained spelling), `Name`, `Args` (`[]*Param` via the shared `parseParamList` — typed and bare args), `Body` (`*string`, raw-string delimiters removed via `stripRawString` **only** — no dedent/trim/unescape), and `HasUnsupportedBody`.
- An empty raw body (`#""#`) is a non-nil pointer to the empty string.
- The historical brace-body form is balanced-skipped for recovery, marked `HasUnsupportedBody`, and **never** fabricated into a `Body` string.

**Grammar/lexer:** reuses the existing participle field grammar + RawString lexer (no Jinja grammar, no hand-rolled block parser). The `Ident` rule is widened for BAML's optional leading `$`, and `string_template` is added to the top-level-keyword stop set.

**Drift guard:** `internal/nativeprompt`'s dynamic-template drift guard now reads `FunctionBlock.PromptRaw` instead of reaching through generic `Fields`. It stays test-only.

### Laxer-than-BAML choices (logged)

Native must never out-claim BAML. Each of the following is deliberately laxer than BAML (which gates invalid input at build), carries a `FIX-EVENTUALLY (#586)` code comment, and is logged on issue #586: the `string_template` alias, the optional leading-`$` identifier, the optional `=` before the template body, and the brace-body tolerance. **No grammar-parity claim is made** — only v0.222 is present in the module cache (non-authoritative); a build-only v0.223 acceptance fixture remains the intended merge guard before any parity claim.

### Tests

Upgraded the existing prompt/raw/template parser tests to assert the dedicated fields, and added coverage for: exact-body (the real `cmd/build/dynamic.baml` prompt, byte-equal after delimiter removal), typed + bare template args, optional-equals, both keyword spellings, multi-hash bodies (function prompt + template body), empty body, CRLF/tabs/blank-line no-trim/no-dedent, last-field-wins (with a final non-raw prompt declining), the brace-tolerated case, and multi-file macro source ordering.

### Validation

- `gofmt` clean; `go build ./...`; `go vet ./...` (+ `-tags=integration`) — no new issues (only pre-existing participle struct-tag warnings).
- `go test ./bamlutils/bamlparser ./internal/nativeschema ./cmd/introspect ./internal/nativeprompt` — all pass.
- Config parser golden suite (`TestBamlConfigGoldens`) byte-identical.
- `go run ./cmd/embed` then `go run ./cmd/regenerate-dynclient` — no manifest/generated diff.
- `(cd workerplugin && GOWORK=off go test ./... -count=1)` — pass.
- `go build ./cmd/introspect` (package-form) — OK.

<a href="https://cubic.dev/pr/invakid404/baml-rest/pull/600?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Parser now exposes raw function prompts via dedicated prompt metadata, using last-prompt-wins while preserving the full original fields order.
  * Template parsing now retains richer template details (keyword, name, args, raw body) and flags unsupported body formats, including correct handling for empty and multi-hash raw strings.
  * Expanded template syntax support (keyword aliasing, optional assignment, and `$`-prefixed identifier args).

* **Bug Fixes**
  * Prevented template declarations from being consumed during type-related scanning.
  * Improved tolerance for legacy brace-bodied templates.

* **Tests**
  * Expanded prompt/body retention, template parsing, ordering, and edge-case coverage (including dynamic prompt drift checks).

### CODE REVIEW

File: bamlutils/bamlparser/bamlparser_test.go [hunk 1]
```diff
@@ -823,3 +875,350 @@ func TestParse_RetryPolicyMalformedNestedTokenBetweenFieldsSkipped(t *testing.T)
...
+			t.Errorf("PromptRaw lost %q", needle)
+		}
+	}
+	if !strings.HasPrefix(*fn.PromptRaw, "\n") {
+		t.Errorf("leading whitespace was trimmed from the prompt body")
+	}
+	if !strings.HasSuffix(*fn.PromptRaw, "\n  ") {
+		t.Errorf("trailing whitespace was trimmed from the prompt body")
+	}
+}
```
[the remark this answers is not in the document]
Comment (invakid404) [reply bamlutils/bamlparser/bamlparser_test.go:1040]: Declining as a low-value test nit: dynamic.baml currently has exactly one function and one raw prompt, and the test already obtains Baml_Rest_Dynamic before comparing. A future preceding prompt would make this check fail unless it were intentionally byte-identical, rather than conceal a parser or parity regression. The added source-scoping logic is not worth a CI cycle for this representation-only slice.
